browser/ui/webui/brave_rewards_page_ui.cc

```diff
@@ -318,13 +320,7 @@ const char kAutoDetectedAdsSubdivisionTargeting[] =
 RewardsDOMHandler::RewardsDOMHandler() : weak_factory_(this) {}
 
 RewardsDOMHandler::~RewardsDOMHandler() {
-  if (rewards_service_) {
-    rewards_service_->RemoveObserver(this);
-  }
-
-  if (ads_service_) {
-    ads_service_->RemoveObserver(this);
-  }
+  OnJavascriptDisallowed();
```

Comment:
Does this need to be called in the destructor? Or is it automatically called from the WebUI machinery? Looking at other message handler classes in Chromium, I see a couple that call this in the destructor, but most do not.

Reply:
Yes it is called on RenderFrameHost unload, so this is probably redundant.